Svc/SeqDispatcher/SeqDispatcher.cpp

```diff
@@ -0,0 +1,204 @@
+// ======================================================================
+// \title  SeqDispatcher.cpp
+// \author root
+// \brief  cpp file for SeqDispatcher component implementation class
+// ======================================================================
+
+#include <FpConfig.hpp>
+#include <components/fsw/SeqDispatcher/SeqDispatcher.hpp>
+
+namespace components {
+
+// ----------------------------------------------------------------------
+// Construction, initialization, and destruction
+// ----------------------------------------------------------------------
+
+SeqDispatcher ::SeqDispatcher(const char* const compName)
+    : SeqDispatcherComponentBase(compName) {}
+
+SeqDispatcher ::~SeqDispatcher() {}
+
+NATIVE_INT_TYPE SeqDispatcher::getNextAvailableSequencerIdx() {
+  for (int i = 0; i < types::CMD_SEQUENCERS_COUNT; i++) {
+    if (this->m_entryTable[i].state == types::CmdSequencerState::AVAILABLE) {
+      return i;
+    }
+  }
+  return -1;
+}
+
+bool SeqDispatcher::runSequence(NATIVE_INT_TYPE sequencerIdx,
+                                const Fw::String& fileName,
+                                Svc::CmdSequencer_BlockState block) {
+  FW_ASSERT(sequencerIdx >= 0 && sequencerIdx < types::CMD_SEQUENCERS_COUNT,
+            sequencerIdx);
+
+  // if the seqRunOut port at this idx isn't connected
+  if (!this->isConnected_seqRunOut_OutputPort(sequencerIdx)) {
+    this->log_WARNING_HI_INVALID_SEQUENCER(static_cast<U16>(sequencerIdx));
```

Comment:
Instead of checking if the port is connected on use, it might make sense to check if the sequence engine is connected on initialization and asserting if not - letting users know if the topology is misconfigured as quickly as possible.

Reply:
Great call! Will do

Reply:
Code written, needs further testing

Reply:
Looking through this more, I think the right answer is add an isConnected call to the `getNextAvailableSequenceIdx`.

That way users could connect up as many or as few sequencers as they want in their pool.  The "catch" for users who did not connect-up sequencers would be to look for unconnected ports in the topology.

That way a project could use two sequencer pools of different sizes if they wanted, without needing to connect every port.